### ISSUE TITLE
Using workflows: No notification when new article was created

### ISSUE THREAD

**Issue by alve89** [joomla/joomla-cms#35320]
User A (author) creates a new article and sets the default stage _Zur Prüfung einreichen_ (he can't choose anything else)
![image](https://user-images.githubusercontent.com/11487054/130470233-1d7c3945-2c8a-4ef5-b258-25af3943dabd.png)

Then the article has the default stage. Since no transition was triggered, no notification was sent to any users. How would publishers know that there are new articles without any notification?

J4.0.0

**Comment by simbus82** [comment]
What you see in the Workflow selector IS a transition. Check that you have enabled notification in the transition settings
![image](https://user-images.githubusercontent.com/1375730/130472621-2228ec3c-5711-4995-8ab4-ce89711552e1.png)

**Comment by alve89** [comment]
No, what I see is not a transition, but the default stage (_Zur Prüfung einreichen_):
![image](https://user-images.githubusercontent.com/11487054/130582478-80e32a0e-3159-4553-94d5-221fa392c2a4.png)

What is possible is to allow executing a transition for a user group:
![image](https://user-images.githubusercontent.com/11487054/130582699-7080e150-fdaa-4c08-9c36-b18c3c647cbb.png)

But I can't make a stage inaccessible for a user group:
![image](https://user-images.githubusercontent.com/11487054/130582921-d872b578-2063-4f53-bca7-9b6c4fc356f8.png)

So if I would allow a transition to a user group, the default stage would always be choosable as well:
![image](https://user-images.githubusercontent.com/11487054/130583077-8ac2db71-02de-4c33-898b-d662d44b5919.png)

Copied from #35300 from @bembelimen:
> Creating an article is not using the workflow. If the article is created in the frontend, then you can setup the Joomla! plugin to inform admins. If the article is created in the backend, there is no messaging will a transition is executed. (But that didn't change from 3.x...)


That's a problem I think. A workflow - in general, not (yet) in Joomla - starts at the very beginning, which would be, related to Joomla, the creation of an article.

Without deeper knowledge of the workflow concept, an imaginable scenario would be to set a default transition similar to a default stage. 

Since stages can be set without any transitions but not vice-versa, the process could be like this:
1. If default transition is set, default stage is ignored (default transition is automatically set to ACL executable)
2. Since default stage is ignored, only default transition (and, depending on permissions, more transitions) is visible in the articles dropdown field

Maybe my suggestion of a solution isn't working but at the moment we're still at the point of discussing where/when a workflow is starting (or: should be starting) - so let's start here and let's get the same understanding of this.

**Comment by bembelimen** [comment]
You can execute a first transition the moment you create the article (if the transition is available).

We thought to add the possibility to define "default transitions", which are executed by default, but so far it hadn't the highest priority.

Feel free to spend some thought there and probably come up with a solution.

**Comment by mrownicki** [comment]
I test recently in 4.0.4 Workflow in create myself + turn on notification for 2 users. Several transitions.
I confirm notification. Notification via mail not incoming.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35320">issues.joomla.org/tracker/joomla-cms/35320</a>.</sub>
